lrlex/src/lib/ctbuilder.rs

```diff
@@ -79,6 +80,26 @@ pub enum RustEdition {
     Rust2021,
 }
 
+pub trait LexErrorHandler<LexerTypesT>
+where
+    LexerTypesT: LexerTypes,
+    usize: num_traits::AsPrimitive<LexerTypesT::StorageT>,
+{
+    /// Called with the lexers filename
+    fn lexer_path(&mut self, filename: &Path);
+    /// Called with the lexers source contents.
+    fn lexer_src(&mut self, src: &str);
```

Comment:
one thing that just came to mind is we could consider adding here is `fn generated_src(&mut self, src:&str)`?